teps/0091-trusted-resources.md

```diff
@@ -364,26 +373,110 @@ type KeylessRef struct {
+3) Skip the verification if no policies exist and `verification-no-match-policy` is set to `allow`.
+
+**Note:** The current proposed `verification-no-match-policy` will replace the old `resource-verification-mode` in one release and this is not a backwards compatible change.
+
+ **How configmap works with Verification Policy**
+ * If the `verification-no-match-policy` is set to `deny`
+   * No Matching Policies: Resources that do not have matched policies will fail the taskrun/pipelinerun.
+   * Matching Policies:
+     * Don't fail taskrun/pipelinerun if all the matched `enforce` policies pass or only match `warn` policies.
+     * Fail the taskrun/pipelinerun if any matched `enforce` policy fails to verify.
+
+ * If the `verification-no-match-policy` is set to `allow` or `warn`,
+   * No Matching Policies: Don't fail taskrun/pipelinerun, and will emit warning log if set to `warn`.
+   * Matching Policies: Same as `deny`.
+
+
+ #### Condition Update
+
+ Trusted resources should also update the taskrun/pipelinerun’s condition to indicate if it passes verification or not. This can be done via knative api. Other options are listed in [alternatives](#alternatives).
+
+ Need to create conditions for trusted resources so this can co-exist with current `apis.ConditionSucceeded` type:
+ ```go
+ const (
+	// ConditionVerification specifies that the resources pass trusted resources verification or not.
+	ConditionVerification apis.ConditionType = "TrustedResourcesVerification"
+)
+ ```
+
+Conditions by Case:
+
+1. All matching policies pass verification
+Condition:
+ ```go
+ apis.Condition{
+			Type:    ConditionVerification,
+			Status:  corev1.ConditionTrue,
+			Reason:  podconvert.ReasonResourceVerificationSuccess
+			Message: "Trusted resource verification passed",
+		}
+ ```
+
+Resulting yaml status:
+```yaml
+status:
+  completionTime: "2023-03-01T18:17:10Z"
+  conditions:
+  - lastTransitionTime: "2023-03-01T18:17:05Z"
+    message: Trusted resource verification passed
+    reason: ResourceVerificationSucceeded
+    status: "True"
+    type: TrustedResourcesVerification
+  - lastTransitionTime: "2023-03-01T18:17:10Z"
+    message: All Steps have completed executing
+    reason: Succeeded
+    status: "True"
+    type: Succeeded
+```
+
+
+ 2. Policy fails verification.
+Condition:
+ ```go
+ apis.Condition{
+			Type:    ConditionVerification,
+			Status:  corev1.ConditionFalse,
+			Reason:  podconvert.ReasonResourceVerificationFailed
+			Message: "", //filled with error message,
+		}
+ ```
+
+Resulting yaml status:
+```yaml
+status:
+  completionTime: "2023-03-01T18:17:10Z"
+  conditions:
+  - lastTransitionTime: "2023-03-01T18:17:05Z"
+    message: Trusted resource verification failed
+    reason: ResourceVerificationFailed
+    status: "False"
+    type: TrustedResourcesVerification
+  - lastTransitionTime: "2023-03-01T18:17:10Z"
+    message: All Steps have completed executing
+    reason: Succeeded
+    status: "True"
+    type: Succeeded
```

Comment:
I think I'm revisiting a previous discussion, but if the Task fails verification, shouldn't we fail execution? I can't imagine a scenario where a Task fails verification but we want it to execute anyway

Reply:
Oh that's true, I think for warn mode policy, we shouldn't fail the execution, for enforce mode we should fail. I will update this part